cs3api4lab/handlers.py

```diff
@@ -2,15 +2,114 @@
+from cs3api4lab.api.cs3apismanager import CS3APIsManager
+from cs3api4lab.api.cs3_share_api import Cs3ShareApi
+
+
+class ShareHandler(APIHandler):
+    @property
+    def share_api(self):
+        return Cs3ShareApi(self.log)
+
+    @web.authenticated
+    @gen.coroutine
+    def post(self):
+        request = self.get_json_body()
+        RequestHandler.handle_request(self,
+                                      self.share_api.create,
+                                      201,
+                                      request['endpoint'],
+                                      request['file_id'],
+                                      request['grantee'],
+                                      request['idp'],
+                                      request['role'],
+                                      request['grantee_type'])
+
+    @web.authenticated
+    @gen.coroutine
+    def delete(self):
+        share_id = self.get_query_argument('share_id')
+        RequestHandler.handle_request(self, self.share_api.remove, 204, share_id)
+
+    def put(self):
+        request = self.get_json_body()
+        RequestHandler.handle_request(self,
+                                      self.share_api.update,
+                                      204,
+                                      request['share_id'],
+                                      request['role'])
+
+
+class ListSharesHandler(APIHandler):
+    @property
+    def share_api(self):
+        return Cs3ShareApi(self.log)
 
     @web.authenticated
     @gen.coroutine
     def get(self):
-        output = {
-            'hello': 'world'
-        }
-        self.set_header('Content-Type', 'application/json')
-        self.finish(json.dumps(output))
+        RequestHandler.handle_request(self, self.share_api.list, 200)
+
+
+class ListReceivedSharesHandler(APIHandler):
+    @property
+    def share_api(self):
+        return Cs3ShareApi(self.log)
+
+    @web.authenticated
+    @gen.coroutine
+    def get(self):
+        RequestHandler.handle_request(self, self.share_api.list_received, 200)
+
+    @web.authenticated
+    @gen.coroutine
+    def put(self):
+        share_id = self.get_query_argument('share_id', default=None)
+        state = self.get_query_argument('state', default='pending')
+        RequestHandler.handle_request(self, self.share_api.update_received, 200, share_id, state)
+
 
-handlers = [(r"/api/cs3/helloworld", HelloWorldHandle)]
\ No newline at end of file
+class ListSharesForFile(APIHandler):
+    @property
+    def share_api(self):
+        return Cs3ShareApi(self.log)
+
+    @web.authenticated
+    @gen.coroutine
+    def get(self):
+        file_id = self.get_query_argument('file_id', default=None)
+        RequestHandler.handle_request(self, self.share_api.list_grantees_for_file, 200, file_id)
+
+
+handlers = [
+    (r"/api/cs3/shares", ShareHandler),
+    (r"/api/cs3/shares/list", ListSharesHandler),
+    (r"/api/cs3/shares/received", ListReceivedSharesHandler),
```

Comment:
According to your documentation, this should be `list-received`. Please change one or the other.